app/src/main/java/com/github/se/signify/ui/screens/home/ASLRecognition.kt

```diff
@@ -35,117 +35,98 @@ import com.github.se.signify.model.home.hand.HandLandmarkViewModel
 import com.github.se.signify.model.navigation.NavigationActions
 import com.github.se.signify.ui.common.AnnexScreenScaffold
 import com.github.se.signify.ui.common.CameraBox
+import com.github.se.signify.ui.common.LetterDictionary
 import com.github.se.signify.ui.common.TextButton
-import com.google.mediapipe.tasks.components.containers.NormalizedLandmark
 
 /**
  * Composable that handles ASL recognition. It checks for camera permissions, launches the camera
  * preview, and displays recognized gestures and images.
  */
-@OptIn(ExperimentalMaterial3Api::class)
 @Composable
 fun ASLRecognition(
     handLandmarkViewModel: HandLandmarkViewModel,
     navigationActions: NavigationActions
 ) {
   val buttonUriString = stringResource(id = R.string.button_uri_string_text)
   val context = LocalContext.current
+
+  val coroutineScope = rememberCoroutineScope()
+
+  val landmarksState = handLandmarkViewModel.landMarks().collectAsState()
+  val detectedGesture = handLandmarkViewModel.getSolution()
+
+  val imageResource =
+      if (landmarksState.value.isNullOrEmpty()) R.drawable.vector
+      else getIconResId(detectedGesture.first())
+
+  val makeASignText = stringResource(id = R.string.make_a_sign_text)
+  val displayText = if (landmarksState.value.isNullOrEmpty()) makeASignText else detectedGesture
+  val paintColor = MaterialTheme.colorScheme.background
   AnnexScreenScaffold(navigationActions = navigationActions, testTag = "ASLRecognitionScreen") {
     Box(
         modifier =
-            Modifier.fillMaxWidth()
-                .height(252.dp)
-                .padding(horizontal = 16.dp)
-                .background(MaterialTheme.colorScheme.background)) {
+            Modifier.fillMaxWidth().weight(1f).background(MaterialTheme.colorScheme.background)) {
           CameraBox(handLandmarkViewModel, "cameraPreview")
         }
 
-    Spacer(modifier = Modifier.height(30.dp))
+    Spacer(modifier = Modifier.height(16.dp))
 
-    GestureOverlayView(handLandmarkViewModel)
+    LetterDictionary(
+        coroutineScope = coroutineScope,
+        numbOfHeaders = integerResource(R.integer.scroll_offset),
+        onClick = { _, _ -> })
 
-    Spacer(modifier = Modifier.height(20.dp))
-    val moreOnASLAlphabetText = stringResource(id = R.string.more_on_asl_alphabet_text)
+    Spacer(modifier = Modifier.height(16.dp))
+
+    // Draws the output text or the prompt to "Make a sign" on the screen.
```

Comment:
There is two duplicate comments I don't think it is relevant.
Also you can create here a function named : 
`HandGestureOverlay() {}` that contains the Output/"Make a sign" text. It will help reviewing it and make it easy for maintenance and modularization.